### DISCUSSION TITLE
Windows Update for CH340 driver breaking USB serial for fake chips

### DISCUSSION THREAD

**Discussion by thickbut** [awawa-dev/HyperHDR#582] [General]
The crux of it is explained here: https://github.com/SHWotever/SimHub/wiki/Arduino---Counterfeit-Fake-CH340G-chips-driver-issues
If anyone uses USB serial to send data to HyperHDR, then you may have experienced connection issues if you don't have a genuine USB to serial chip on your Arduino/ESP device. This is because Windows pushed an update for CH340 drivers for the first time in 4 years. I had this problem for 2 months and ended up using UDP via WLED because I didn't know what was wrong. The device selection in the HyperHDR menu and logs were still identifying the correct port and fake chip information but It just wouldn't connect. I rolled back the drivers and now it works again. But in the future, Windows Update will probably reinstall the new driver and break my device again.

**Comment by awawa-dev** [comment]
Hi
Thanks for the info about a new Windows driver that is causing problems with some counterfeit CH340s. Unfortunately, for a very long time we encountered the problem of counterfeit CH340 even on Linux, but usually it was enough to lower the speed to make it work (but of course it's better to just choose the original product without the shortcomings or recommended esp32(-s2)/RaspberryPiPico as an alternative)
